File: 7-kyu/going-to-the-cinema/javascript/solution.js
const movie = (card, ticket, perc) => {
  let ret = 1;
  let card_price = card + ticket * perc;
  let total_ticket = ticket;
  while (!(total_ticket > Math.ceil(card_price))) {
    ret += 1;
    card_price += ticket * Math.pow(perc, ret);
    total_ticket += ticket
  }
  return ret;
};

File: 7-kyu/going-to-the-cinema/swift/solution.swift
func movie(card: Double, ticket: Double, perc: Double) -> Int {
    var ret = 1
    var card_price: Double = card + ticket * perc
    var total_ticket = Int(ticket)
    while !(Int(ceil(card_price)) < total_ticket) {
        ret += 1
        card_price += ticket * (pow(perc, Double(ret)))
        total_ticket += Int(ticket)
    }
    return ret
}
